### DISCUSSION TITLE
Place label in a center of a white, irregular area

### DISCUSSION THREAD

**Discussion by ranknfyle** [ImageMagick/ImageMagick#6312] [Help]
### ImageMagick version

6.8.9

### Operating system, version and so on

Win10, Perl 5.16

### Description

I have a big black canvas. I then read a sequence of images, same size as the canvas. They're mostly black, but each contains a single, white splotch. I composite all splotches in the single big image. The splotches don't overlap. I want to place a label in the center of each white splotch, but I don't know how to determine the coordinates of the center of a splotch. I could use a Trim on each individual splotch image, to get the bounding box of the splotch and find its center. But this doesn't give me the offset to correctly place the label in the big image. How can I do this? I'm on old version of IM (6.8.9) with Perl (5.16), but I don't think it matters.

![splotches2](https://user-images.githubusercontent.com/5205381/236962685-cad70ef6-439f-47ea-9ff9-b6c97956e198.png)

**Comment by snibgo** [comment]
I suggest: for each splotch image, make non-white pixels transparent, then write the label on the image. Then you can composite each labelled splotch on the black canvas.

Does that help?

**Reply by ranknfyle** [reply]
Thanks for the reply, but how does making black pixels transparent help me place a label in the center of the white splotch? Compositing or labeling is not an issue. The issue is finding the coordinates of the center of each splotch, so a label can be placed in that exact spot. Each splotch image is the same size as the final composite.

**Comment by snibgo** [comment]
The transparency may help the final composite.

I'm not sure what problem you have. As you say, you can trim each splodge and write the label at its centre.

I suggest you post sample inputs and the command you have so far.

**Comment by fmw42** [comment]
Post your input image without labels.  One can use -connected-components to find the centroids and then write the labels to the centroids.

**Reply by ranknfyle** [reply]
Thanks I'll try "Connected Components"

**Reply by fmw42** [reply]
You don't need it if you have separate images with only one white region in each image. You can simple get the trim coordinate (bounding box) and then compute the center of the bounding box and put your label there.  You can write a script FOR loop over each image and get the center that way. Or use the solution on StackOverflow from Mark Setchell.
